llvm/lib/Transforms/IPO/RepoPruning.cpp

```diff
@@ -105,6 +106,42 @@ ticketmd::DigestType toDigestType(pstore::index::digest D) {
   return Digest;
 }
 
+static void addDependentFragments(
+    Module &M, StringSet<> &DependentFragments,
+    std::shared_ptr<const pstore::index::fragment_index> const &Fragments,
+    const pstore::database &Repository, pstore::index::digest const &Digest) {
+
+  auto It = Fragments->find(Repository, Digest);
+  assert(It != Fragments->end(Repository));
+  // Create  the dependent fragments if existing in the repository.
+  auto Fragment = pstore::repo::fragment::load(Repository, It->second);
+  if (auto Dependents =
+          Fragment->atp<pstore::repo::section_kind::dependent>()) {
+    for (pstore::typed_address<pstore::repo::compilation_member> Dependent :
+         *Dependents) {
+      auto CM = pstore::repo::compilation_member::load(Repository, Dependent);
+      StringRef MDName =
+          toStringRef(pstore::get_sstring_view(Repository, CM->name).second);
+      LLVM_DEBUG(dbgs() << "    Prunning dependent name: " << MDName << '\n');
+      auto DMD =
+          TicketNode::get(M.getContext(), MDName, toDigestType(CM->digest),
+                          toGVLinkage(CM->linkage), true);
+      // If functions 'A' and 'B' are dependent on function 'C', only add a
+      // single TicketNode of 'C' to the 'repo.tickets' in order to avoid
+      // multiple compilation_members of function 'C' in the compilation.
```

Comment:
As we discussed F2F, the function 'addDependentFragments' cover the cases:
a) A->B  B->C
b) A->B->C
But you added comments only for the (a) case.
Can you add extra comments for the (b) case. Otherwise the function would imply that covers only case (a).

Reply:
Fixed the commnet, thanks.